olive/cli/auto_opt.py

```diff
@@ -215,8 +220,140 @@ def _get_data_config(self) -> List[Dict]:
+    def _get_passes_config(self, passes_config: Dict[str, Any], olive_config: OlivePackageConfig) -> Dict[str, Any]:
+        if self.args.mixed_precision_overrides_config and len(self.args.mixed_precision_overrides_config) % 2 != 0:
+            raise ValueError("Even number of entries required for mixed precision overrides config.")
+
+        mixed_precision_overrides_config = (
+            {
+                self.args.mixed_precision_overrides_config[i]: self.args.mixed_precision_overrides_config[i + 1]
+                for i in range(0, len(self.args.mixed_precision_overrides_config), 2)
+            }
+            if self.args.mixed_precision_overrides_config
+            else None
+        )
+
+        to_replace = [
+            (("bnb4", "quant_type"), PRECISION_MAPPING["bnb4"].get(self.args.precision, self.args.precision)),
+            (
+                ("dynamic_quant", "weight_type"),
+                PRECISION_MAPPING["dynamic_quant"].get(self.args.precision, self.args.precision),
+            ),
+            (
+                ("model_builder", "precision"),
+                PRECISION_MAPPING["model_builder"].get(self.args.precision, self.args.precision),
+            ),
+            (("to_fixed_shape", "dim_param"), self.args.dynamic_to_fixed_shape_dim_param),
+            (("to_fixed_shape", "dim_value"), self.args.dynamic_to_fixed_shape_dim_value),
+            (("mixed_precision_overrides", "overrides_config"), mixed_precision_overrides_config),
+        ]
+        for keys, value in to_replace:
+            if value is not None:
+                set_nested_dict_value(passes_config, keys, value)
+
+        del passes_config["conversion" if self.args.use_model_builder else "model_builder"]
+        if self.args.provider != "DmlExecutionProvider":
+            # Use the DynamicToFixedShape pass only for DmlExecutionProvider
+            # becase Direct ML doesn't support dynamic shaped inputs
+            del passes_config["to_fixed_shape"]
+
+        for pass_name in list(passes_config.keys()):
+            pass_run_config = passes_config[pass_name]
+            pass_module_config = olive_config.get_pass_module_config(pass_run_config["type"])
+            if (
+                (self.args.precision not in pass_module_config.supported_precisions)
+                or (self.args.provider not in pass_module_config.supported_providers)
+                or (self.args.device not in pass_module_config.supported_accelerators)
+            ):
+                del passes_config[pass_name]
+
+        if "to_fixed_shape" in passes_config and not (
+            self.args.dynamic_to_fixed_shape_dim_param and self.args.dynamic_to_fixed_shape_dim_value
+        ):
+            raise ValueError("dim-param and dim-value are required for DynamicToFixedShape.")
+
+        if "conversion" not in passes_config and "model_builder" not in passes_config:
+            raise ValueError("Cannot export an onnx model with combination of provided options.")
```

Comment:
What if input model is an ONNX model

Reply:
Needed to handle the "metadata_only" scenario!